src/dex/balancer-v1/balancer-v1.ts

```diff
@@ -0,0 +1,729 @@
+import { Interface } from '@ethersproject/abi';
+import { Contract } from 'web3-eth-contract';
+import { DeepReadonly } from 'ts-essentials';
+import BigNumber from 'bignumber.js';
+import _ from 'lodash';
+import * as bmath from '@balancer-labs/sor/dist/bmath';
+import {
+  Address,
+  ExchangePrices,
+  Log,
+  AdapterExchangeParam,
+  SimpleExchangeParam,
+  PoolLiquidity,
+  Logger,
+  Token,
+  PoolPrices,
+} from '../../types';
+import { SwapSide, Network, MAX_UINT_BIGINT } from '../../constants';
+import { StatefulEventSubscriber } from '../../stateful-event-subscriber';
+import {
+  wrapETH,
+  getDexKeysWithNetwork,
+  isETHAddress,
+  interpolate,
+  bignumberify,
+} from '../../utils';
+import { IDex } from '../../dex/idex';
+import { IDexHelper } from '../../dex-helper/idex-helper';
+import {
+  BalancerV1Data,
+  PoolState,
+  DexParams,
+  PoolStateMap,
+  OptimizedBalancerV1Data,
+  BalancerParam,
+  BalancerFunctions,
+  PoolStates,
+  Token as SORToken,
+} from './types';
+import { SimpleExchange } from '../simple-exchange';
+import {
+  BalancerV1Config,
+  Adapters,
+  defaultfactoryAddress,
+  defaultMulticallAddress,
+  LogCallTopics,
+  poolUrls,
+  POOL_FETCH_TIMEOUT,
+  SUBGRAPH_TIMEOUT,
+  BALANCER_CHUNKS,
+  BALANCER_SWAP_GAS_COST,
+} from './config';
+import BalancerV1ABI from '../../abi/BalancerV1.json';
+import BalancerCustomMulticallABI from '../../abi/BalancerCustomMulticall.json';
+import { AxiosResponse } from 'axios';
+import { Pool as OldPool } from '@balancer-labs/sor/dist/types';
+import { calcInGivenOut, calcOutGivenIn } from '@balancer-labs/sor/dist/bmath';
+import { typecastReadOnlyPool } from './utils';
+import { parsePoolPairData } from './sor-overload';
+
+const balancerV1Interface = new Interface(BalancerV1ABI);
+
+export class BalancerV1EventPool extends StatefulEventSubscriber<PoolStateMap> {
+  handlers: {
+    [event: string]: (event: any, pool: PoolState, log: Log) => PoolState;
+  } = {};
+
+  logDecoder: (log: Log) => any;
+
+  public allPools: PoolState[] = [];
+
+  addressesSubscribed: string[];
+
+  balancerMulticall: Contract;
+
+  constructor(
+    protected parentName: string,
+    protected network: number,
+    protected dexHelper: IDexHelper,
+    logger: Logger,
+    protected adapters = Adapters[network] || {},
```

Comment:
this is not used anywhere

Reply:
I thought `adapters` is a part of public unterface. It is used in `getAdapters`. The same is in `UniswapV2`. Do you think it is better to remove it completely?

Reply:
Removed adapters